app/code/core/Mage/Eav/Model/Entity/Collection/Abstract.php

```diff
@@ -167,9 +167,7 @@ protected function _prepareStaticFields()
     protected function _initSelect()
     {
         $this->getSelect()->from(array('e' => $this->getEntity()->getEntityTable()));
-        if ($this->getEntity()->getTypeId()) {
-            $this->addAttributeToFilter('entity_type_id', $this->getEntity()->getTypeId());
-        }
+
```

Comment:
I believe this will accomplish the same but still keep compatibility with the default entity table (which is not used in core but is still present).

```suggestion
        if ($this->getEntity()->getEntityTable() != Mage_Eav_Model_Entity::DEFAULT_ENTITY_TABLE && $this->getEntity()->getTypeId()) {
            $this->addAttributeToFilter('entity_type_id', $this->getEntity()->getTypeId());
        }
```